beacon-chain/core/blocks/block_operations.go

```diff
@@ -105,6 +72,40 @@ func verifyBlockRandao(proposer *pb.ValidatorRecord, block *pb.BeaconBlock) erro
 	return nil
 }
 
+// ProcessEth1Data is an operation performed on each
+// beacon block to ensure the ETH1 data votes are processed
+// into the beacon state.
+//
+// Official spec definition of ProcessEth1Data
+//   If block.eth1_data equals eth1_data_vote.eth1_data for some eth1_data_vote
+//   in state.eth1_data_votes, set eth1_data_vote.vote_count += 1.
+//   Otherwise, append to state.eth1_data_votes a new Eth1DataVote(eth1_data=block.eth1_data, vote_count=1).
+func ProcessEth1Data(beaconState *pb.BeaconState, block *pb.BeaconBlock) (*pb.BeaconState, error) {
+	var eth1DataVoteAdded bool
+
+	for _, Eth1DataVote := range beaconState.Eth1DataVotes {
+		if bytes.Equal(Eth1DataVote.Eth1Data.BlockHash32, block.Eth1Data.BlockHash32) {
+			if bytes.Equal(Eth1DataVote.Eth1Data.DepositRootHash32, block.Eth1Data.DepositRootHash32) {
```

Comment:
Please reduce nesting by putting these conditions in the same if statement.

```go
if bytes.Equal(Eth1DataVote.Eth1Data.BlockHash32, block.Eth1Data.BlockHash32) && bytes.Equal(Eth1DataVote.Eth1Data.DepositRootHash32, block.Eth1Data.DepositRootHash32) {
...
}
```

Reply:
👍